api_tests/steps/check_oauth.py

```diff
@@ -29,4 +32,22 @@ def get_token_response(self, timeout: int = 5000, grant_type: str = 'authorizati
             data['_access_token_expiry_ms'] = timeout
 
         response = self.post(self.endpoints['token'], data=data)
-        return self.get_all_values_from_json_response(response)
\ No newline at end of file
+        return self.get_all_values_from_json_response(response)
+
+    def post(self, url: str, **kwargs) -> 'response type':
```

Comment:
There is no need for a separate `post` method. It does more harm than good. As Ben mentioned it'll mask the root cause of exception